src/storage/mod.rs

```diff
@@ -665,13 +666,13 @@ impl<E: Engine> Storage<E> {
                             ScanMode::Forward,
                             options.key_only,
                             Some(start_key),
-                            None,
+                            end_key,
                         )?;
                     } else {
                         scanner = snap_store.scanner(
                             ScanMode::Backward,
                             options.key_only,
-                            None,
+                            end_key,
```

Comment:
It is a little weird to let start_key as upper bound.

Reply:
We have some mixing of "start_key" in the code base. The semantic here is correct, means the key to start scanning, so in backward it is upper bound and in forward scanning it is lower bound. However we also use start_key in other places that doesn't have a direction, i.e. region range. They should be renamed to a more correct one indicating an order.